File: 백준/Gold/11657. 타임머신/타임머신.py
# 타임머신

import sys
input = sys.stdin.readline

N, M = map(int, input().split())
arr = []
minRoute = [sys.maxsize] * (N + 1)
minRoute[1] = 0

for _ in range(M):
    s, e, w = map(int, input().split())
    arr.append((s, e, w))

for _ in range(N - 1):
    for s, e, w in arr:
        if minRoute[s] != sys.maxsize and minRoute[e] > minRoute[s] + w:
            minRoute[e] = minRoute[s] + w

success = True

for s, e, w in arr:
    if minRoute[s] != sys.maxsize and minRoute[e] > minRoute[s] + w:
        success = False

if success:
    for value in minRoute[2:]:
        if value == sys.maxsize:
            print(-1)
        else:
            print(value)
else:
    print(-1)

File: 백준/Gold/11657. 타임머신/타임머신.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, M] = input.shift().split(' ').map(Number);
const edges = input.map(line => line.split(' ').map(Number));

const distance = Array(N + 1).fill(Infinity);
distance[1] = 0;

let cycle = false; // 사이클 존재 여부를 체크하는 변수

for (let i = 1; i <= N; i++) {
  for (let j = 0; j < M; j++) {
    const [from, to, cost] = edges[j];
    if (distance[from] !== Infinity && distance[to] > distance[from] + cost) {
      distance[to] = distance[from] + cost;
      if (i === N) {
        cycle = true; // N번째 라운드에서도 갱신이 일어난다면 음의 사이클이 존재하는 것
        break;
      }
    }
  }
  if (cycle) break; // 음의 사이클이 감지되면 루프를 빠져나옴
}

if (cycle) {
  console.log(-1);
} else {
  for (let i = 2; i <= N; i++) {
    if (distance[i] === Infinity) {
      console.log(-1);
    } else {
      console.log(distance[i]);
    }
  }
}
